analyzers/src/SonarAnalyzer.CSharp/Facade/CSharpSyntaxFacade.cs

```diff
@@ -66,24 +72,26 @@ node switch
                 InvocationExpressionSyntax invocation => invocation.Expression,
                 LockStatementSyntax @lock => @lock.Expression,
                 null => null,
-                _ => throw Unexpected(node)
+                _ => throw InvalidOperation(node, nameof(NodeExpression))
             };
 
         public override SyntaxToken? NodeIdentifier(SyntaxNode node) =>
-            node switch
+            RemoveParentheses(node) switch
             {
+                AssignmentExpressionSyntax assignment => NodeIdentifier(assignment.Left),
```

Comment:
This line is in a common code and makes sense only for this particular rule. The root cause of this problem needs to be fixed on the invocation side. Mainly because there could be other unexpected syntaxes that could lead that path to be reached.
```suggestion
```

Reply:
Fare enough, but I have no clue how this is triggered. Do you?

Reply:
I thought the last code snippet will reproduce it. If I comment the `CSharpSyntaxFacade` line, I get the exception on these two:
```
result = (localVariable = 42) != -1);

if ((localVariable = 42) != -1) { }
```

Reply:
Now it does. 
``` C#
result = (localVariable = 42) == 0;
result = 0 == (localVariable = 42);
```

Do not evaluate as alwaysTrue of alwaysFalse. The question is, how to deal with this. I agree that the first attempt has downsides, which leaves us to the following options:
 1. return null for Assigenments
 2. catch the exception
 3. have an HasIdentifier
 4. Something I can not think of right now

And I dislike 1,2, and 3 for different reasons.

Reply:
I'd go for option 4 😄 

Update `NodeIdentifier` from
```
null => null,
_ => throw InvalidOperation(node, nameof(NodeIdentifier)),
```
to
```
_ => null,
```

Background reasoning:
* `HasIdentifier` option is the best idea. And implementing it is effectively the same as `NodeIdentifier != null` except the exception
* It's hard to predict what will arrive in `NodeIdentifier`, we can accept it as a fact.
* Returning `null` from the method is already expected behavior on many paths

Reply:
I think that was option 1. I'll do that. Also reconsidered that being the best option.

Reply:
I understood the option 1 as `AssignmentExpressionSyntax  => null`

Reply:
@pavel-mikula-sonarsource I understand that your read that into it, but that was not what I intended. Anyhow. I changed the implementation as agreed on, and now, all tests pass.